cuenca/http/client.py

```diff
@@ -137,6 +115,31 @@ def request(
         self._check_response(resp)
         return resp.json()
 
+    def set_jwt_headers(self):
+        # Make sure the current token is still valid
+        try:
+            payload_encoded = self.jwt_token.split('.')[1]
+            payload = json.loads(base64.b64decode(f'{payload_encoded}=='))
+            # Get a new token if there's less than 5 mins for the actual
+            # to be expired
+            if dt.datetime.utcfromtimestamp(
+                payload['exp']
+            ) - dt.datetime.utcnow() <= dt.timedelta(minutes=5):
+                raise Exception('Expired token')
```

Comment:
huh? why take this approach?

Reply:
Payload in JWT tokens contain `exp` in base64 which indicates the datetime when the token will be invalid. So I'm extracting that part of the token to validate it before I send the request, this way I save one request saying that the token is expired.